# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/glibc23/gcc33-hotfix.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

Rene Rebe: This fixes compilation with gcc-3.3
Clifford: And the soinit.c thing is a workaroung for an x86_64 ICE

--- glibc-2.3.2/stdio-common/sscanf.c	2002-08-10 20:09:08.000000000 +0200
+++ glibc-2.3.2/stdio-common/sscanf.c	2003-05-13 09:46:19.000000000 +0200
@@ -27,12 +27,10 @@
 /* Read formatted input from S, according to the format string FORMAT.  */
 /* VARARGS2 */
 int
-sscanf (s, format)
-     const char *s;
-     const char *format;
+sscanf (const char *s, const char *format, ... )
 {
-  va_list arg;
   int done;
+  va_list arg;
 
   va_start (arg, format);
   done = __vsscanf (s, format, arg);
--- ./elf/soinit.c.orig	Tue Feb 24 15:55:00 2004
+++ ./elf/soinit.c	Tue Feb 24 16:31:08 2004
@@ -25,7 +25,7 @@
 }
 
 #ifdef HAVE_DWARF2_UNWIND_INFO
-static char __EH_FRAME_BEGIN__[]
+const static char __EH_FRAME_BEGIN__[]
      __attribute__ ((section (".eh_frame")))
      = { };
 # ifdef HAVE_DWARF2_UNWIND_INFO_STATIC
